modules/javafx.media/src/main/java/com/sun/media/jfxmedia/Media.java

```diff
@@ -98,7 +97,7 @@ public List<Track> getTracks() {
             if (tracks.isEmpty()) {
                 returnValue = null;
             } else {
-                returnValue = Collections.unmodifiableList(new ArrayList<>(tracks));
+                returnValue = List.copyOf(tracks);
             }
         }
         return returnValue;
```

Comment:
This method can be reduced to
```java
    public List<Track> getTracks() {
        synchronized(tracks) {
            return tracks.isEmpty() ? null : List.copyOf(tracks);
        }
    }
```
though I find it highly questionable that it returns `null` for an empty list instead of just an empty list. There are 2 use cases of this method and both would do better with just an empty list.

Reply:
Yeah, I noticed this as well right away, it is documented to do this though.  The documentation however does seem to suggest it might be possible that there are three results:

1. Tracks haven't been scanned yet -> `null`
2. Tracks have been scanned, but none where found -> empty list
3. Tracks have been scanned and some were found -> list

Whether case 2 can ever happen is unclear, but distinguishing it from the case where nothing has been scanned yet with `null` does not seem unreasonable.

Reply:
It's an internal class and no calling class makes this distinction. I don't think it's meant to function in the way you described.